packages/twenty-server/src/database/commands/upgrade-version-command/upgrade.command.ts

```diff
@@ -21,28 +20,35 @@ import { SyncWorkspaceMetadataCommand } from 'src/engine/workspace-manager/works
   name: 'upgrade',
   description: 'Upgrade workspaces to the latest version',
 })
-export class UpgradeCommand extends ActiveOrSuspendedWorkspacesMigrationCommandRunner {
+export class UpgradeCommand extends UpgradeCommandRunner {
+  fromVersion = new SemVer('0.43.0');
+
   constructor(
     @InjectRepository(Workspace, 'core')
     protected readonly workspaceRepository: Repository<Workspace>,
+    protected readonly environmentService: EnvironmentService,
     protected readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    protected readonly syncWorkspaceMetadataCommand: SyncWorkspaceMetadataCommand,
+
+    // Commands injection
     protected readonly migrateRichTextContentPatchCommand: MigrateRichTextContentPatchCommand,
     protected readonly addTasksAssignedToMeViewCommand: AddTasksAssignedToMeViewCommand,
     protected readonly migrateIsSearchableForCustomObjectMetadataCommand: MigrateIsSearchableForCustomObjectMetadataCommand,
     protected readonly updateDefaultViewRecordOpeningOnWorkflowObjectsCommand: UpdateDefaultViewRecordOpeningOnWorkflowObjectsCommand,
     protected readonly migrateSearchVectorOnNoteAndTaskEntitiesCommand: MigrateSearchVectorOnNoteAndTaskEntitiesCommand,
-    protected readonly syncWorkspaceMetadataCommand: SyncWorkspaceMetadataCommand,
+    ///
   ) {
-    super(workspaceRepository, twentyORMGlobalManager);
-  }
-
-  override async runOnWorkspace(args: RunOnWorkspaceArgs): Promise<void> {
-    this.logger.log(
-      chalk.blue(
-        `${args.options.dryRun ? '(dry run)' : ''} Upgrading workspace ${args.workspaceId} ${args.index + 1}/${args.total}`,
-      ),
+    super(
+      workspaceRepository,
+      environmentService,
+      twentyORMGlobalManager,
+      syncWorkspaceMetadataCommand,
     );
+  }
 
+  override async beforeSyncMetadataUpgradeCommandsToRun(
+    args: RunOnWorkspaceArgs,
+  ) {
     await this.migrateRichTextContentPatchCommand.runOnWorkspace(args);
 
     await this.migrateIsSearchableForCustomObjectMetadataCommand.runOnWorkspace(
```

Comment:
update this upgrade command for 0.44. We should only have the initialize-permission

Reply:
Will do in an other PR